### ISSUE TITLE
v8_0_17 cannot find fill_help_tables.sql

### ISSUE THREAD

**Issue by jlabanca** [wix/wix-embedded-mysql#181]
When using v8_0_17, I get the following error:

```
de.flapdoodle.embed.process.exceptions.DistributionException: java.nio.file.NoSuchFileException: C:\Users\John\.embedmysql\extracted\Windows-B64--\MySQL-8.0\mysql-8.0.17\share\fill_help_tables.sql
	at de.flapdoodle.embed.process.runtime.Starter.prepare(Starter.java:69)
	at de.flapdoodle.embed.process.runtime.Starter.prepare(Starter.java:49)
	at com.wix.mysql.EmbeddedMysql.<init>(EmbeddedMysql.java:41)
	at com.wix.mysql.EmbeddedMysql$Builder.start(EmbeddedMysql.java:169)
	at com.syndicloud.jsonsync.store.MySqlJsonSyncStoreTest.<init>(MySqlJsonSyncStoreTest.java:48)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:217)
	at org.junit.runners.BlockJUnit4ClassRunner$1.runReflectiveCall(BlockJUnit4ClassRunner.java:266)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)
Caused by: java.nio.file.NoSuchFileException: C:\Users\John\.embedmysql\extracted\Windows-B64--\MySQL-8.0\mysql-8.0.17\share\fill_help_tables.sql
	at sun.nio.fs.WindowsException.translateToIOException(Unknown Source)
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
	at sun.nio.fs.WindowsException.rethrowAsIOException(Unknown Source)
	at sun.nio.fs.WindowsFileCopy.copy(Unknown Source)
	at sun.nio.fs.WindowsFileSystemProvider.copy(Unknown Source)
	at java.nio.file.Files.copy(Unknown Source)
	at de.flapdoodle.embed.process.extract.ExtractedFileSets.copy(ExtractedFileSets.java:62)
	at de.flapdoodle.embed.process.store.ExtractedArtifactStore.extractFileSet(ExtractedArtifactStore.java:110)
	at com.wix.mysql.store.SafeExtractedArtifactStore.extractFileSet(SafeExtractedArtifactStore.java:30)
	at de.flapdoodle.embed.process.runtime.Starter.prepare(Starter.java:57)
	... 26 more
```

fill_help_tables.sql doesn't appear in the .embedmysql\extracted\Windows-B64--\MySQL-8.0\mysql-8.0.17\share folder.

**Comment by viliusl** [comment]
@jlabanca - can you try `8.0.11` - I see it works on our CI (https://ci.appveyor.com/project/viliusl/wix-embedded-mysql/build/job/ninjdkch1peslmgm#L605)

Will investigate 8.0.17 issues, but unable to do it quickly, so maybe `8.0.11` will unblock you

**Comment by jlabanca** [comment]
I ran into a separate issue with 8.0.11:
https://github.com/wix/wix-embedded-mysql/issues/180

But I'm using v5_7_27 for now so I'm not blocked.

**Comment by lsoares** [comment]
same here:

```
[main] INFO com.wix.mysql.EmbeddedMysql - Preparing EmbeddedMysql version 'Version 8.0.17'...
[main] ERROR de.flapdoodle.embed.process.runtime.Starter - no message
java.nio.file.NoSuchFileException: C:\Users\Luís Soares\.embedmysql\extracted\Windows-B64--\MySQL-8.0\mysql-8.0.17\share\fill_help_tables.sql
```

**Comment by soaserele** [comment]
We had the same issue with MySQL v 8.0.17. After comparing the MySQL archives from "https://dev.mysql.com/downloads/mysql/" we found that `share\fill_help_tables.sql` file is missing from the archive. Also `bin/resolveip` and `share/mysql_system_tables*` files are also missing. We had to patch the code to get wix-embedded-mysql running for MySQL v 8.0.17. 

[mysql8_17_fileset.patch.txt](https://github.com/wix/wix-embedded-mysql/files/3770427/mysql8_17_fileset.patch.txt)

P.S. To get it running on Linux, we also had to install the following libraries:
```
libaio1 libnuma1 libssl1.0 libcrypto++6
```

**Comment by viliusl** [comment]
@jlabanca @soaserele - turns out 8.0.11 and 8.0.17 had quite some differences and deprecations. I added both versions to version test matrix (8.0.17 was not in), win/osx/nix tests pass, released version `4.6.1`.

Closing now - if smth still does not work please reopen/let me know.
